qiskit/circuit/delay.py

```diff
@@ -97,26 +87,25 @@ def duration(self, duration):
         self.params = [duration]
 
     def to_matrix(self) -> np.ndarray:
-        """Return a Numpy.array for the unitary matrix. This has been
-        added to enable simulation without making delay a full Gate type.
-
-        Returns:
-            np.ndarray: matrix representation.
-        """
-        return self.__array__(dtype=complex)
+        """Return a Numpy.array for the unitary matrix."""
+        return np.eye(2**self.num_qubits, dtype=complex)
```

Comment:
Here also: please don't delete the explanatory comment.

This also _should_ keep the definition of the `__array__` function.  `to_matrix` (for some reason) is just a wrapper around that method, and inherited from the parent class.